mmcls/models/backbones/vig.py

```diff
@@ -0,0 +1,600 @@
+            build_norm_layer(dict(type='BN'), out_dim)[1],
+        )
+
+    def forward(self, x):
+        x = self.convs(x)
+        return x
+
+
+@MODELS.register_module()
+class Vig(BaseBackbone):
+    # n_blocks,channels
+    arch_settings = {'tiny': [12, 192], 'small': [16, 320], 'base': [16, 640]}
+
+    def __init__(self,
+                 arch,
+                 k=9,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='BN'),
+                 graph_conv_bias=True,
+                 graph_conv_type='mr',
+                 epsilon=0.2,
+                 use_dilation=True,
+                 use_stochastic=False,
+                 drop_path=0.,
+                 relative_pos=False,
+                 norm_eval=False,
+                 frozen_stages=0,
+                 init_cfg=None):
+        super().__init__(init_cfg=init_cfg)
+        arch = self.arch_settings[arch]
+        self.n_blocks = arch[0]
+        channels = arch[1]
+        self.stem = Stem(out_dim=channels, act=act_cfg)
+        dpr = [x.item() for x in torch.linspace(0, drop_path, self.n_blocks)
+               ]  # stochastic depth decay rule
+        num_knn = [
+            int(x.item()) for x in torch.linspace(k, 2 * k, self.n_blocks)
+        ]  # number of knn's k
+        max_dilation = 196 // max(num_knn)
+        self.norm_eval = norm_eval
+        self.pos_embed = nn.Parameter(torch.zeros(1, channels, 14, 14))
+        self.frozen_stages = frozen_stages
+        if use_dilation:
+            self.stage_blocks = Sequential(*[
+                Sequential(
+                    Grapher(
+                        channels,
+                        num_knn[i],
+                        min(i // 4 + 1, max_dilation),
+                        graph_conv_type,
+                        act_cfg,
+                        norm_cfg,
+                        graph_conv_bias,
+                        use_stochastic,
+                        epsilon,
+                        1,
+                        drop_path=dpr[i],
+                        relative_pos=relative_pos),
+                    FFN(channels, channels * 4, act=act_cfg, drop_path=dpr[i]))
+                for i in range(self.n_blocks)
+            ])
+        else:
+            self.stage_blocks = Sequential(*[
+                Sequential(
+                    Grapher(
+                        channels,
+                        num_knn[i],
+                        1,
+                        graph_conv_type,
+                        act_cfg,
+                        norm_cfg,
+                        graph_conv_bias,
+                        use_stochastic,
+                        epsilon,
+                        1,
+                        drop_path=dpr[i],
+                        relative_pos=relative_pos),
+                    FFN(channels, channels * 4, act=act_cfg, drop_path=dpr[i]))
+                for i in range(self.n_blocks)
+            ])
```

Comment:
Since the only difference between `use_dilation=True` and `use_dilation=False` is the `dilation` argument, directly use:
```suggestion
        dilation = min(i // 4 + 1, max_dilation) if use_dilation else 1
        self.stage_blocks = Sequential(*[
            Sequential(
                Grapher(
                    in_channels=channels,
                    k=num_knn[i],
                    dilation=dilation,
                    conv=conv_graph_conv_type,
                    act=act_cfg,
                    norm=norm_cfg,
                    bias=graph_conv_bias,
                    stochastic=use_stochastic,
                    epsilon=epsilon,
                    drop_path=dpr[i],
                    relative_pos=relative_pos),
                FFN(channels, channels * 4, act=act_cfg, drop_path=dpr[i]))
            for i in range(self.n_blocks)
        ])
```

Reply:
And better to use keyword arguments instead of positional arguments, so that it's easier for users to get the relationship.

Reply:
Ok, I will change it.